### PR TITLE
Infrastructural changes for Telemetry Info Generator

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
This PR contains **infrastructure** for a telemetry information generator program. The program parses through the state field registry after the construction of the main control loop and lists, in JSON format, all of the readable/writable state fields on the satellite and their serialization parameters. The goal of this program is to make it easier to perform our telemetry review in January.

### CODE REVIEW

**Comment by tanishqaggarwal** [comment]
Example output, generated by the script:

`{"fields":{"adcs.ang_rate":{"bitsize":4,"max":10.0,"min":0.0,"type":"float","writable":false},"adcs.cmd_attitude":{"bitsize":29,"type":"float quaternion","writable":true},"adcs.min_stable_ang_rate":{"bitsize":4,"max":10.0,"min":0.0,"type":"float","writable":true},"adcs.mode":{"bitsize":4,"max":10,"min":0,"type":"unsigned char","writable":true},"adcs_box.mag_vec":{"bitsize":96,"max":1.0,"min":0.0,"type":"float vector","writable":false},"adcs_box.sun_vec":{"bitsize":96,"max":1.0,"min":0.0,"type":"float vector","writable":false},"attitude_estimator.q_body_eci":{"bitsize":29,"type":"float quaternion","writable":false},"attitude_estimator.w_body":{"bitsize":96,"max":55.0,"min":-55.0,"type":"float vector","writable":false},"docksys.config_cmd":{"bitsize":1,"type":"bool","writable":true},"docksys.dock_config":{"bitsize":1,"flow_id":2,"type":"bool","writable":false},"docksys.docked":{"bitsize":1,"flow_id":2,"type":"bool","writable":false},"docksys.is_turning":{"bitsize":1,"flow_id":2,"type":"bool","writable":false},"downlink_producer.radio_err_ptr":{"bitsize":7,"max":10,"min":-90,"type":"signed int","writable":false},"gomspace.battmode":{"bitsize":3,"max":4,"min":0,"type":"unsigned char","writable":false},"gomspace.bootcause":{"bitsize":4,"max":8,"min":0,"type":"unsigned char","writable":false},"gomspace.counter_boot":{"bitsize":32,"max":4294967295,"min":0,"type":"unsigned int","writable":false},"gomspace.counter_wdt_i2c":{"bitsize":32,"max":4294967295,"min":0,"type":"unsigned int","writable":false},"gomspace.curin.output1":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curin.output2":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curin.output3":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output1":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output2":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output3":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output4":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output5":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.curout.output6":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.cursun":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.cursys":{"bitsize":10,"max":10,"min":0,"type":"unsigned int","writable":false},"gomspace.output.output1":{"bitsize":1,"type":"bool","writable":false},"gomspace.output.output2":{"bitsize":1,"type":"bool","writable":false},"gomspace.output.output3":{"bitsize":1,"type":"bool","writable":false},"gomspace.output.output4":{"bitsize":1,"type":"bool","writable":false},"gomspace.output.output5":{"bitsize":1,"type":"bool","writable":false},"gomspace.output.output6":{"bitsize":1,"type":"bool","writable":false},"gomspace.pptmode":{"bitsize":1,"max":2,"min":1,"type":"unsigned char","writable":false},"gomspace.temp.output1":{"bitsize":8,"max":125,"min":-40,"type":"signed int","writable":false},"gomspace.temp.output2":{"bitsize":8,"max":125,"min":-40,"type":"signed int","writable":false},"gomspace.temp.output3":{"bitsize":8,"max":125,"min":-40,"type":"signed int","writable":false},"gomspace.temp.output4":{"bitsize":8,"max":125,"min":-40,"type":"signed int","writable":false},"gomspace.vbatt":{"bitsize":10,"max":9000,"min":5000,"type":"unsigned int","writable":false},"gomspace.vboost.output1":{"bitsize":9,"max":4000,"min":0,"type":"unsigned int","writable":false},"gomspace.vboost.output2":{"bitsize":9,"max":4000,"min":0,"type":"unsigned int","writable":false},"gomspace.vboost.output3":{"bitsize":9,"max":4000,"min":0,"type":"unsigned int","writable":false},"gomspace.wdt_i2c_time_left":{"bitsize":10,"max":1000,"min":0,"type":"unsigned int","writable":false},"pan.cycle_no":{"bitsize":32,"max":4294967295,"min":0,"type":"unsigned int","writable":false},"pan.deployed":{"bitsize":1,"flow_id":1,"type":"bool","writable":false},"pan.mode":{"bitsize":4,"flow_id":1,"max":10,"min":0,"type":"unsigned char","writable":true},"pan.sat_designation":{"bitsize":2,"flow_id":1,"max":2,"min":0,"type":"unsigned char","writable":true},"piksi.baseline_pos":{"bitsize":100,"max":100000.0,"min":0.0,"type":"double vector","writable":false},"piksi.pos":{"bitsize":100,"max":100000.0,"min":0.0,"type":"double vector","writable":false},"piksi.state":{"bitsize":2,"max":4,"min":0,"type":"unsigned int","writable":false},"piksi.time":{"bitsize":68,"type":"gps_time_t","writable":false},"piksi.vel":{"bitsize":100,"max":100000.0,"min":0.0,"type":"double vector","writable":false}},"flows":[{"active":true,"fields":["pan.mode","pan.deployed","pan.sat_designation"],"id":1,"priority":0},{"active":true,"fields":["docksys.docked","docksys.dock_config","docksys.is_turning"],"id":2,"priority":1}]}`

**Comment by tanishqaggarwal** [comment]
Adding @shihaocao so he can look at the Piksi changes.

File: lib/Drivers/Piksi.hpp [hunk 1]
```diff
@@ -408,7 +408,7 @@ class Piksi {
     bool _user_data_update;
 
     //set read return mock
-    #ifdef DESKTOP
+    #if defined(DESKTOP) || defined(UNIT_TEST) 
     unsigned int _read_return;
     #endif
 };
```
Comment (shihaocao) [inline lib/Drivers/Piksi.hpp]: Will debug task eventually deal with _read_return somehow? If we only have the DESKTOP flag set, it'll always return 0 for read_all(), which corresponds to SPP

Comment (tanishqaggarwal) [reply lib/Drivers/Piksi.hpp]: I think we could just set the Piksi mode via psim, yeah.

File: lib/Drivers/Piksi.hpp [hunk 2]
```diff
@@ -408,7 +408,7 @@ class Piksi {
     bool _user_data_update;
 
     //set read return mock
-    #ifdef DESKTOP
```
Comment (athena255) [inline lib/Drivers/Piksi.hpp]: This breaks hardware tests

**Pull request merged**